src/porepy/numerics/ad/operators.py

```diff
@@ -181,6 +261,23 @@ def is_leaf(self) -> bool:
         """
         return len(self.children) == 0
 
+    @property
+    def is_original_operator(self) -> bool:
```

Comment:
Base class has now a check to see if the operator represents the current time and iter.

Reply:
Sure.
This name is based on previous code, where the TimeDependentArray class had an attribute `original_operator`
How about `is_current_time` or `is_current_iter`. The latter one probably is more precise, because current iter is always current time.

Reply:
Sounds reasonable (but I'd personally prefer "iteration" or "iterate"). On that, though. What exactly is the benefit of having `TimeDependentOperator` separate from `IterativeOperator`?

Reply:
Looking for example at the current TImeDependentARRAY.
It has obviously a behavior in time.
But it can also be part of multiple operators at different iterates, always returning the value corresponding to its time index.

Reply:
I see (and thanks for the capitalization of Array! I'm sure I would've been confused without it).

Reply:
Changed to `is_current_iterate`